lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/GroupCommClient.cs

```diff
@@ -71,18 +73,21 @@ public sealed class GroupCommClient : IGroupCommClient
             }
 
             networkService.Register(new StringIdentifier(taskId));
+        }
 
+        public void WaitingForRegistration(CancellationTokenSource cancellationSource = null)
+        {
             try
             {
                 foreach (var group in _commGroups.Values)
                 {
-                    group.WaitingForRegistration();
+                    group.WaitingForRegistration(cancellationSource);
                 }
             }
-            catch (ReefRuntimeException e)
+            catch (SystemException e)
```

Comment:
Shall we un-inherited from ReefRuntimeException for thsoe two exceptions in a separate Jira?